src/metpy/io/station_data.py

```diff
@@ -193,6 +193,9 @@ def key_finder(df):
 
     df['latitude'] = None
     df['longitude'] = None
+    df = df.astype({'latitude': np.float64,
+                    'longitude': np.float64})
```

Comment:
```suggestion
    df['latitude'] = np.nan
    df['longitude'] = np.nan
```

With no other changes, we should be able to just init these with `nan`, and the Series should automatically start with dtype float. This `astype` conversion interprets `None` as nan anyway.

Reply:
I like this approach.